### ISSUE TITLE
Network requests fail with 400 only in Chrome browser (not in Electron)

### ISSUE THREAD

**Issue by ashutoshchoudhary7** [cypress-io/cypress#8401]
### Current behavior:
Using cypress we are launching our application "https://www.bigwphotos.com.au/" in chrome browse and performing below operation:

1. Click on Canvas & Wall Decor
2. Click on first product "12 x 18" (30x45cm) Canvas Print"
3. Click on "Shop Now"

We are getting redirected to page not found error "get product error Request failed with status code 400"
![Product Not available](https://user-images.githubusercontent.com/40062322/91144891-439b6700-e6d2-11ea-930a-764548041897.JPG)

We are getting this only in chrome browser in electron it is executing as per desired behavior
### Desired behavior:

Using cypress after launching our application "https://www.bigwphotos.com.au/" in chrome browser and performing below operation

1. Click on Canvas & Wall Decor
2. Click on first product "12 x 18" (30x45cm) Canvas Print"
3. Click on "Shop Now"

We should be redirected to image upload page
We are able to navigate to image upload manually and with other automation framework however getting blocked in cypress
![Expected](https://user-images.githubusercontent.com/40062322/91145050-7d6c6d80-e6d2-11ea-8252-56df076988c5.JPG)
### Test code to reproduce

```js
describe('My First Cypress Test', function() {
    Cypress.on('uncaught:exception', (err, runnable) => {
        // returning false here prevents Cypress from
        // failing the test
        return false
      })
    it('Visits the BigWPhotos Page and check the menu items', function() {
    //Visit the BigWPhotos Website
    cy.visit('https://www.bigwphotos.com.au/');
    //Click on Canvas&Wall decor
    cy.get(".desk-menu > :nth-child(4) > [href='/collections/canvas-wall-decor']", { timeout: 10000 })
    .should('be.visible');
    cy.xpath('(.//a[@title="Canvas & Wall Decor"])[1]').click({force: true});
    //(//*[@*='thumb'])[1]
    cy.xpath('(.//*[@*="thumb"])[1]', { timeout: 10000 })
    .should('be.visible');
    cy.get(":nth-child(1) > .productGridItem > .details-container > .details > strong > .productMainLink").click({force: true});
    cy.get('#AddToCartText', { timeout: 10000 })
    .should('be.visible');
    cy.get('#AddToCartText').click({force: true});
    cy.get('[class="LoadingIndicator__Wrapper-hodrYc hwrwkA LoadingIndicator"]', { timeout: 10000 })
    .should('be.visible');
    cy.get('[class="LoadingIndicator__Wrapper-hodrYc hwrwkA LoadingIndicator"]', { timeout: 10000 })
    .should('not.be.visible');
    cy.wait(10000);
})
})
```

### Versions
Cypress 5.0

**Comment by jennifer-shehane** [comment]
I am seeing some errors regarding same-site cookies, as your application is not handling this correctly in Chrome. Chrome rolled out changes to samesite cookies, but I feel like this should be breaking your application in general if it's an issue - not only within Cypress.

I'm not sure why this behavior is different.

### Repro

```js
Cypress.on('uncaught:exception', () => {
  return false
})

it('Visits the BigWPhotos Page and check the menu items', {
    commandTimeout: 10000
  }, () => {
  cy.visit('https://www.bigwphotos.com.au/collections/canvas-wall-decor/products/12-x-18-30x45cm-canvas-print')
  cy.get('#AddToCartText').click()
  cy.url().should('include', 'productId=1948')
  cy.url().should('include', 'computer')
})
```

### Electron

<img width="1671" alt="Screen Shot 2020-08-26 at 2 19 48 PM" src="https://user-images.githubusercontent.com/1271364/91276206-39976800-e7a7-11ea-9b98-2a2298dc0f5c.png">

Has `origin` request header

<img width="1098" alt="Screen Shot 2020-08-26 at 2 24 04 PM" src="https://user-images.githubusercontent.com/1271364/91276826-0c978500-e7a8-11ea-8583-70b00be43650.png">


### Chrome in Cypress

<img width="1608" alt="Screen Shot 2020-08-26 at 2 18 59 PM" src="https://user-images.githubusercontent.com/1271364/91276141-1f5d8a00-e7a7-11ea-99ea-51d5304e6fc8.png">


<img width="1669" alt="Screen Shot 2020-08-26 at 1 56 56 PM" src="https://user-images.githubusercontent.com/1271364/91275976-dd344880-e7a6-11ea-9cbb-ac8f66b5a26e.png">

<img width="1059" alt="Screen Shot 2020-08-26 at 2 18 22 PM" src="https://user-images.githubusercontent.com/1271364/91276076-0654d900-e7a7-11ea-999b-a30d9027e05f.png">

Request headers look the same except that it's missing the `origin` header.

<img width="1098" alt="Screen Shot 2020-08-26 at 2 24 04 PM" src="https://user-images.githubusercontent.com/1271364/91276860-16b98380-e7a8-11ea-987d-c4624face3b7.png">


### Chrome outside of Cypress

The cookies error also shows for 3 resources. 

<img width="1107" alt="Screen Shot 2020-08-26 at 2 20 36 PM" src="https://user-images.githubusercontent.com/1271364/91276307-5c298100-e7a7-11ea-993a-ed4966c6c55f.png">

**Comment by jennifer-shehane** [comment]
I'm not able to recreate this in the current version of cypress. Please update and let us know if your issue is resolved. We can reopen if not.

**Comment by MasonM** [comment]
@jennifer-shehane I'm encountering a very similar problem and stumbled on this issue. I was able to reproduce this problem in Cypress 7.5.0 using the reproduction code you posted earlier, and I think I know what the problem is. As you correctly stated, [Chrome rolled out changes for cookies](https://www.chromium.org/updates/same-site) that will cause it to default to `SameSite=lax`. When the `SameSite` attribute is set to `lax`, a cookie is only sent for same-site requests or cross-site top-level navigation:
> If the value is "Lax", the cookie will be sent with same-site requests, and with "cross-site" top-level navigations, as described in Section 5.3.7.1.

https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-rfc6265bis-05

The problem lies with cross-site requests. In the reproduction code, the user is taken from https://www.bigwphotos.com.au/ to https://4.app.fujifilmimagine.com/product-redirect?productId=1948, which is a cross-site request. When you do this manually (outside Cypress), it will also count as a top-level navigation, because the URL in the browser bar changes. However, this is not the case for Cypress: only the iframe URL changes, while the top-level URL will stay as https://www.bigwphotos.com.au/. Therefore, the cookie is blocked by Chrome and the test fails.

What this means is that it's impossible to test any functionality in Cypress that relies on cookies with `SameSite=lax` when a cross-site request is made. It's possible to workaround this issue by adding `--disable-features=SameSiteByDefaultCookies` to your Chrome Launch options (as described as [Overcoming SameSite cookie issue in Cypress when running on Chrome or Edge](https://dev.to/poponuts/overcoming-samesite-cookie-issue-in-cypress-when-running-on-chrome-or-edge-274m)), but this flag will be removed in Chrome 94, so we can't rely on this.

Let me know if you'd like me to open a separate issue for this.

**Comment by jennifer-shehane** [comment]
@MasonM cross domain doesn't really work in general in Cypress, so may need to wait for this issue, which we're working on : https://github.com/cypress-io/cypress/issues/944
